In [17]:
import numpy as np
import pandas as pd
import seaborn as sns

In [18]:
sns.get_dataset_names()

['anagrams',
 'anscombe',
 'attention',
 'brain_networks',
 'car_crashes',
 'diamonds',
 'dots',
 'dowjones',
 'exercise',
 'flights',
 'fmri',
 'geyser',
 'glue',
 'healthexp',
 'iris',
 'mpg',
 'penguins',
 'planets',
 'seaice',
 'taxis',
 'tips',
 'titanic']

In [19]:
df = sns.load_dataset('titanic')
df


survived  pclass     sex   age  sibsp  parch     fare embarked   class  \
0           0       3    male  22.0      1      0   7.2500        S   Third   
1           1       1  female  38.0      1      0  71.2833        C   First   
2           1       3  female  26.0      0      0   7.9250        S   Third   
3           1       1  female  35.0      1      0  53.1000        S   First   
4           0       3    male  35.0      0      0   8.0500        S   Third   
..        ...     ...     ...   ...    ...    ...      ...      ...     ...   
886         0       2    male  27.0      0      0  13.0000        S  Second   
887         1       1  female  19.0      0      0  30.0000        S   First   
888         0       3  female   NaN      1      2  23.4500        S   Third   
889         1       1    male  26.0      0      0  30.0000        C   First   
890         0       3    male  32.0      0      0   7.7500        Q   Third   

       who  adult_male deck  embark_town alive  alone  
0      man        True  NaN  Southampton    no  False  
1    woman       False    C    Cherbourg   yes  False  
2    woman       False  NaN  Southampton   yes   True  
3    woman       False    C  Southampton   yes  False  
4      man        True  NaN  Southampton    no   True  
..     ...         ...  ...          ...   ...    ...  
886    man        True  NaN  Southampton    no   True  
887  woman       False    B  Southampton   yes   True  
888  woman       False  NaN  Southampton    no  False  
889    man        True    C    Cherbourg   yes   True  
890    man        True  NaN   Queenstown    no   True  

[891 rows x 15 columns]

In [20]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 15 columns):
 #   Column       Non-Null Count  Dtype   
---  ------       --------------  -----   
 0   survived     891 non-null    int64   
 1   pclass       891 non-null    int64   
 2   sex          891 non-null    object  
 3   age          714 non-null    float64 
 4   sibsp        891 non-null    int64   
 5   parch        891 non-null    int64   
 6   fare         891 non-null    float64 
 7   embarked     889 non-null    object  
 8   class        891 non-null    category
 9   who          891 non-null    object  
 10  adult_male   891 non-null    bool    
 11  deck         203 non-null    category
 12  embark_town  889 non-null    object  
 13  alive        891 non-null    object  
 14  alone        891 non-null    bool    
dtypes: bool(2), category(2), float64(2), int64(4), object(5)
memory usage: 80.7+ KB


In [21]:
df.head()

survived  pclass     sex   age  sibsp  parch     fare embarked  class  \
0         0       3    male  22.0      1      0   7.2500        S  Third   
1         1       1  female  38.0      1      0  71.2833        C  First   
2         1       3  female  26.0      0      0   7.9250        S  Third   
3         1       1  female  35.0      1      0  53.1000        S  First   
4         0       3    male  35.0      0      0   8.0500        S  Third   

     who  adult_male deck  embark_town alive  alone  
0    man        True  NaN  Southampton    no  False  
1  woman       False    C    Cherbourg   yes  False  
2  woman       False  NaN  Southampton   yes   True  
3  woman       False    C  Southampton   yes  False  
4    man        True  NaN  Southampton    no   True

In [23]:
df.isnull().sum()

survived         0
pclass           0
sex              0
age            177
sibsp            0
parch            0
fare             0
embarked         2
class            0
who              0
adult_male       0
deck           688
embark_town      2
alive            0
alone            0
dtype: int64

# In the above table , we can find 177 missing value , 2 missing value in embark_town whereas in deck column we can find 688 missing values.

In [24]:
df.head(7)

survived  pclass     sex   age  sibsp  parch     fare embarked  class  \
0         0       3    male  22.0      1      0   7.2500        S  Third   
1         1       1  female  38.0      1      0  71.2833        C  First   
2         1       3  female  26.0      0      0   7.9250        S  Third   
3         1       1  female  35.0      1      0  53.1000        S  First   
4         0       3    male  35.0      0      0   8.0500        S  Third   
5         0       3    male   NaN      0      0   8.4583        Q  Third   
6         0       1    male  54.0      0      0  51.8625        S  First   

     who  adult_male deck  embark_town alive  alone  
0    man        True  NaN  Southampton    no  False  
1  woman       False    C    Cherbourg   yes  False  
2  woman       False  NaN  Southampton   yes   True  
3  woman       False    C  Southampton   yes  False  
4    man        True  NaN  Southampton    no   True  
5    man        True  NaN   Queenstown    no   True  
6    man        True    E  Southampton    no   True

In [25]:
df.tail(7)

survived  pclass     sex   age  sibsp  parch    fare embarked   class  \
884         0       3    male  25.0      0      0   7.050        S   Third   
885         0       3  female  39.0      0      5  29.125        Q   Third   
886         0       2    male  27.0      0      0  13.000        S  Second   
887         1       1  female  19.0      0      0  30.000        S   First   
888         0       3  female   NaN      1      2  23.450        S   Third   
889         1       1    male  26.0      0      0  30.000        C   First   
890         0       3    male  32.0      0      0   7.750        Q   Third   

       who  adult_male deck  embark_town alive  alone  
884    man        True  NaN  Southampton    no   True  
885  woman       False  NaN   Queenstown    no  False  
886    man        True  NaN  Southampton    no   True  
887  woman       False    B  Southampton   yes   True  
888  woman       False  NaN  Southampton    no  False  
889    man        True    C    Cherbourg   yes   True  
890    man        True  NaN   Queenstown    no   True

In [33]:
df.count()

survived       891
pclass         891
sex            891
age            714
sibsp          891
parch          891
fare           891
embarked       889
class          891
who            891
adult_male     891
deck           203
embark_town    889
alive          891
alone          891
dtype: int64

In [34]:
df.shape

(891, 15)

In [35]:
a , b = df.shape
print(f'In the above table it contain {a} number of rows and {b} number of columns.')

In the above table it contain 891 number of rows and 15 number of columns.


In [44]:
for num in range(0,10):
    if num > 1:
        for item in range(2,num):
            if num % item == 0:
                break
            else:
                print(num)

3
5
5
5
7
7
7
7
7
9


In [1]:
q = 4

In [2]:
type(q)

int

In [3]:
id(q)

2246289850704

# list 
* append,extend,update,sort,reverse,insert,clear,len,count,update,pop,copy,index,len,slicing

In [5]:
l = [1,2,3,4,5] # series of number enclosed in []
l.append(58)
l

[1, 2, 3, 4, 5, 58]

In [9]:
k = [9,7,5,4]
l.extend(k)
l


[1, 2, 3, 4, 5, 58, 9, 7, 5, 4, 9, 7, 5, 4, 9, 7, 5, 4, 9, 7, 5, 4]

In [10]:
a = [1,2,3,4]
m = [3,2,1,6]
m.extend(a)

In [12]:
m

[3, 2, 1, 6, 1, 2, 3, 4]

In [16]:
m.sort( reverse = True)
m

[6, 4, 3, 3, 2, 2, 1, 1]

In [17]:
g = [1,8,7]
v = g.copy()

In [18]:
v

[1, 8, 7]

In [19]:
p = [3,2,1,6,7,8]
p[2] = 'prajwal'
p

[3, 2, 'prajwal', 6, 7, 8]

In [21]:
o = ['m','c','d','k','h']
o.insert(1,2)

In [23]:
o

['m', 2, 'c', 'd', 'k', 'h']

In [24]:
o.clear()
o

[]

In [28]:
# slicing , indexing, len
p = [3,2,1,5,6,7]
p[0:5:2]


[3, 1, 6]

In [29]:
i = [1,3,4,5,2,77,88]
i[5]

77

In [30]:
len(i)

7